src/components/legend/defaults.js

```diff
@@ -57,14 +61,30 @@ module.exports = function legendDefaults(layoutIn, layoutOut, fullData) {
     coerce('bordercolor');
     coerce('borderwidth');
     Lib.coerceFont(coerce, 'font', layoutOut.font);
+
     coerce('orientation');
+    if(containerOut.orientation === 'h') {
+        var xaxis = layoutIn.xaxis;
+        if(xaxis && xaxis.rangeslider && xaxis.rangeslider.visible) {
+            defaultX = 0;
+            defaultXAnchor = 'left';
+            defaultY = 1.1;
+            defaultYAnchor = 'bottom';
+        }
+        else {
```

Comment:
@n-riesco can you add a few jasmine test cases in [`legend_test.js`](https://github.com/plotly/plotly.js/blob/c76cdf84b52de2cab636b569aadb863bef6e5081/test/jasmine/tests/legend_test.js#L12) to test out this logic?